### ISSUE TITLE
Sonic Adventure 2 - The rotating title logo textures are broken when using DX11

### ISSUE THREAD

**Issue by kihato** [flyinghead/flycast#1224]
Platform / OS / Hardware: 
Windows11 / i7-10700F / GTX 1660 SUPER

Github hash: 18f9b84

https://github.com/flyinghead/flycast/assets/14366233/e954b47a-2560-4346-b420-ba39c2357b36

This is related to #1207.
The steps to reproduce are as follows.

- Delete dx11oit_shader_cache.bin in advance.
- Launch flycast and select DX11 and per-strip (per-triangle).
- Launch PowerStone and when you see the rotating Capcom logo, pause flycast and switch to per-pixel.
- Once you cancel the pause, return to the game, pause again, and exit PowerStone with Exit.
- Launch Sonic Adventure 2.

This is a demo of Virtua Athlete 2000.
This was the only time I was able to reproduce it and I was never able to reproduce it again.
![VA_6](https://github.com/flyinghead/flycast/assets/14366233/fe9587c9-b061-439b-aa48-b756e1b08f9c)

**Comment by flyinghead** [comment]
Thanks for the detailed steps to reproduce!
This is indeed the same issue as #1207 and the root cause is Flycast changing the host CPU rounding mode to match the guest SH4. This affects how dx11 shaders are compiled.
So I avoid changing the rounding mode on the UI/render thread and set it to default before calling the renderer in single-threaded mode.
Should be fixed in the next master build.

**Comment by kihato** [comment]
I noticed something about switching from per-strip to per-pixel when using DX11.

Virtua athlete 2000 may look like the image below when switching within the demo.
(This is only a momentary flicker.)
![vlcsnap-2023-10-07-08h38m35s721](https://github.com/flyinghead/flycast/assets/14366233/f18dea22-21a2-4ac7-92e3-0d0b7031bafa)

In Sonic Adventure 2, if you switch during a scene where stars are displayed across the screen, dots like the image below may appear on the boot logo.
(I haven't seen this in any games other than SA2, and I also saw it when using flycast v1.3.)
![Swirl_2](https://github.com/flyinghead/flycast/assets/14366233/5eb1b0c2-c6ff-4416-9d4c-033eda4c39c3)

In any case, there doesn't seem to be any particular problem, so I'll close this issue.
Thanks for your quick response.
